### ISSUE TITLE
Writing using hyperslabs is extremely slow compared to netCDF-C

### ISSUE THREAD

**Issue by abhibaruah** [Unidata/netcdf4-python#1222]
netCDF4 version = 1.6.1
netCDF4 hdf5libversion = 1.12.1
netCDF4 netcdf4libversion = 4.8.1

OS: Windows 10

I am trying to create a netCDF4 file and write to it using hyperslabs. However, the writing is extremely slow. The repro code that I posted below takes over 100 seconds. The same workflow using netCDF-C takes a maximum of 10 seconds.

In the repro code below, if I change the 'stride' and 'count' to be (1,1,1) and (400,400,400) respectively, the repro takes < 10 sec.

Is this expected with hyperslab writing or some kind of performance bug?

```
# Import required packages
from netCDF4 import Dataset
from time import perf_counter
import numpy as np
  
# Define file name and variable parameters
filename = 'tempfile_python.nc'
varName = 'xlrg_var'

# Variable parameters
start = (0,0,0)
stride = (2,2,2)
count = (200,200,200)
varSize = (400,400,400)

inpData = np.round(np.random.rand(count[0],count[1],count[2]),3)
      
# Start counter
tInit = perf_counter()
  
# Create file and write data to the variable
rootgrp = Dataset(filename,"w",format="NETCDF4")

  
# Create variable and add data using hyperslab parameters
dimObj_400 = rootgrp.createDimension("dim_400",varSize[2])
inpDims = ("dim_400","dim_400","dim_400")
varObj = rootgrp.createVariable(varName,"f8",inpDims)
(varObj[start[0]:varSize[0]:stride[0],start[1]:varSize[1]:stride[1],
        start[2]:varSize[2]:stride[2]]) = inpData
    
# Close file
rootgrp.close()
  
# Stop counter and return recorded time
execTime = perf_counter() - tInit
print("===Time taken:%f===" % (execTime))


```

**Comment by jswhit** [comment]
No need for the C code @abhibaruah.  PR #1224 should solve it - if you can give it a try and verify that it does that would be great.

**Comment by jswhit** [comment]
closed by PR #1124
